docs/develop-resources/api-conventions/validations.md

```diff
@@ -80,6 +80,8 @@ One future improvement can be having generate-types detect the “Immutable.”
 
 Parent field should be required and immutable.
 
+Note that parent is not a field name, it refers to the cloud resources parent segment in the URL. For example, for GKE clusters, the [GET request URL](https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1/projects.zones.clusters/get) `https://container.googleapis.com/v1/projects/{projectId}/zones/{zone}/clusters/{clusterId}` has `projectId` and `zone` in the parent segment. Therefore, for GKE clusters, projectRef and zone will be the parent fields.
+
```

Comment:
Let's explain "by code" (in the below session, we give a parent example.

Reply:
I do think the code below does a very good job explaining how the parent field is set. I only added this explanation here, IMO is more about how to identify the parent fields of a resource. I had issues correctly identifying and understanding the parent fields.

Reply:
ok, fair enough.